src/phoenix/datasets/experiments.py

```diff
@@ -184,36 +188,29 @@ async def async_run_experiment(test_case: TestCase) -> ExperimentRun:
         if payload is not None:
             resp = client.post(f"/v1/experiments/{experiment_id}/runs", json=jsonify(payload))
             resp.raise_for_status()
-    return Experiment(
+
+    experiment = Experiment(
         id=experiment_id,
         dataset_id=dataset.id,
         dataset_version_id=dataset.version_id,
     )
 
+    if evaluators:
+        for evaluator in evaluators:
+            _evaluate_experiment(experiment, evaluator, dataset.examples)
+
+    return experiment
+
 
-def evaluate_experiment(
+def _evaluate_experiment(
```

Comment:
I think I'm okay exposing this but wonder if this method should ultimately contain the `evaluators` list interface

Reply:
the client can be passed along too

Reply:
what if I make the examples optional so it'll pull them if they aren't passed along, it's kind of an awkward interface but is quick to implement

Reply:
that makes sense- basically let it pass along the experiment cache. Same goes for the client